eng/pipelines/templates/jobs/smoke-tests.yml

```diff
@@ -0,0 +1,168 @@
+parameters:
+  - name: Nightly
+    type: boolean
+    default: true
+  - name: ServiceDirectory
+    type: string
+    default: ""
+  - name: Artifacts
+    type: object
+    default: []
+
+jobs:
+  - ${{ if and(eq(parameters.Nightly, false) }}:
+    - job: smoke_test_eligibility
```

Comment:
Is the goal here to try and prevent all the agents for each matrix entry spinning up to find the same result of no tests to run?

I assume this same logic is included in the smoke test ps1 scripts we should consider a way to refactor that so that we don't forget to maintain one or the other.

Reply:
No, this step is to avoid running smoke tests for releases which aren't touched at all by the smoke tests.

Reply:
Sort of, though at a higher level so that we don't have any misleading jobs in the release to make someone think smoke tests are actually running.

Reply:
Perhaps we could reconcile the logic with the similar logic in the scripts, and have a `whatif` mode or something, so that the code lives in one place, but the pipeline generation is sensible as well?

Reply:
I'm ok with running this job as an upfront job to avoid triggering all the other jobs, however a whatif sort of option would make be feel a little better to make sure the selection logic remains consistent.

Reply:
I added a `-DryRun` switch to the Initialize-SmokeTests.ps1 script, so it will still generate the run-manifest.json as usual, but skip the resource deployment. Then in the job here we just check the package names in the run manifest. It's much cleaner now.